lib/fog/aws/storage.rb

```diff
@@ -256,21 +256,25 @@ def request_params(params)
           if params[:bucket_name]
             bucket_name = params[:bucket_name]
 
-            path_style = params.fetch(:path_style, @path_style)
-            if !path_style
-              if COMPLIANT_BUCKET_NAMES !~ bucket_name
-                Fog::Logger.warning("fog: the specified s3 bucket name(#{bucket_name}) is not a valid dns name, which will negatively impact performance.  For details see: http://docs.amazonwebservices.com/AmazonS3/latest/dev/BucketRestrictions.html")
-                path_style = true
-              elsif scheme == 'https' && !path_style && bucket_name =~ /\./
-                Fog::Logger.warning("fog: the specified s3 bucket name(#{bucket_name}) contains a '.' so is not accessible over https as a virtual hosted bucket, which will negatively impact performance.  For details see: http://docs.amazonwebservices.com/AmazonS3/latest/dev/BucketRestrictions.html")
-                path_style = true
+            if params[:bucket_cname]
+              host = bucket_name
```

Comment:
Bah, sorry. I think maybe I'm just totally mis-remembering. Have you tried using the `endpoint` param? I think maybe setting that to your CNAME might be the right thing. Sorry for the run around here, it's been ages since I interacted with this directly.